etl/nixpkgs-graph.nix

```diff
@@ -1,17 +1,38 @@
 # usage:
-# NIXPKGS_FLAKE_REF="github:nixos/nixpkgs/master" nix eval --json --file "./nixpkgs-graph.nix"
+# NIXPKGS_ALLOW_BROKEN=1 NIXPKGS_ALLOW_INSECURE=1 TARGET_FLAKE_REF="github:nixos/nixpkgs/master" TARGET_FLAKE_SYSTEM="x86_64-linux" nix eval --json --file "./nixpkgs-graph.nix"
 
 let
-  nixpkgsFlakeRef = builtins.getEnv "NIXPKGS_FLAKE_REF";
-  pkgs = import (builtins.getFlake nixpkgsFlakeRef) {
-    config = {
-      # package 'python-2.7.18.6' is marked as insecure
-      # we need to allow this package otherwise it is refusing to evaluate
-      permittedInsecurePackages = [
-        "python-2.7.18.6"
-      ];
-    };
-  };
+  pkgs = builtins.getFlake "nixpkgs";
+  targetFlakeRef = builtins.getEnv "TARGET_FLAKE_REF";
+  # We specify the system of the target flake here to avoid ambiguity
+  # An output will produce a different result on every system
+  targetFlakeSys = builtins.getEnv "TARGET_FLAKE_SYSTEM";
+  targetFlakePkgs =
+    if (builtins.getFlake targetFlakeRef).outputs ? packages
+    then
+      if (builtins.getFlake targetFlakeRef).outputs.packages ? ${targetFlakeSys}
+      then
+        (builtins.getFlake targetFlakeRef).outputs.packages.${targetFlakeSys}
+      else
+        { }
+    else
+      if (builtins.getFlake targetFlakeRef).outputs ? defaultPackage
+      then
+        if (builtins.getFlake targetFlakeRef).outputs.defaultPackage ? ${targetFlakeSys}
+        then
+          (builtins.getFlake targetFlakeRef).outputs.defaultPackage.${targetFlakeSys}
+        else
+          { }
+      else
+        if (builtins.getFlake targetFlakeRef).outputs ? legacyPackages
+        then
+          if (builtins.getFlake targetFlakeRef).outputs.legacyPackages ? ${targetFlakeSys}
+          then
+            (builtins.getFlake targetFlakeRef).outputs.legacyPackages.${targetFlakeSys}
+          else
+            { }
+        else
+          { };
```

Comment:
Could you use a `let ... in` to refactor the many `builtins.getFlake targetFlakeRef`?

e.g.

```nix
let
  flake = builtins.getFlake targetFlakeRef;
in
  if flake.outputs ? packages
  then
    ...
```

Reply:
yes that makes sense!